### ISSUE TITLE
Shallow.debug() does not output a React.memo(React.forwardRef) SFC properly

### ISSUE THREAD

**Issue by Dilatorily** [enzymejs/enzyme#2181]
### Current behavior
The `.debug()` method of a `shallow()` component that is wrapped in both `memo` and `forwardRef`, will return a `Memo([Object object])` component.

This [repository](https://github.com/Dilatorily/enzyme-issue) reproduces the issue.

#### Results

```
 enzyme
    ✓ debugs correctly with a shallow memo'ed SFC (7ms)
    ✓ debugs correctly with a shallow forwarded ref SFC (1ms)
    ✕ debugs correctly with a shallow memo'ed forwarded ref SFC (3ms)

  ● enzyme › debugs correctly with a shallow memo'ed forwarded ref SFC

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `enzyme debugs correctly with a shallow memo'ed forwarded ref SFC 1`

    - Snapshot
    + Received

      "<div>
    -   <Memo(ForwardRef) />
    +   <Memo([object Object]) />
      </div>"

      51 |     };
      52 |     expect(shallow(<Container />).debug())
    > 53 |       .toMatchInlineSnapshot(
         |        ^
      54 |     `
      55 |       "<div>
      56 |         <Memo(ForwardRef) />

      at Object.toMatchInlineSnapshot (index.spec.jsx:53:8)

 › 1 snapshot failed.
```
### Expected behavior
The `debug()` method returns `Memo(ForwardRef)`.

#### API

- [x] shallow
- [ ] mount
- [ ] render

#### Version

| library             | version
| ------------------- | -------
| enzyme              | 3.10.0
| react               | 16.8.6
| react-dom           | 16.8.6
| react-test-renderer | 16.8.6
| adapter (below)     | 1.14.0

#### Adapter

- [x] enzyme-adapter-react-16
- [ ] enzyme-adapter-react-16.3
- [ ] enzyme-adapter-react-16.2
- [ ] enzyme-adapter-react-16.1
- [ ] enzyme-adapter-react-15
- [ ] enzyme-adapter-react-15.4
- [ ] enzyme-adapter-react-14
- [ ] enzyme-adapter-react-13
- [ ] enzyme-adapter-react-helper
- [ ] others ( )

**Comment by Izkata** [comment]
I'm also seeing the same thing without the extra "memo", getting debug output like `<[Object object] />`.  It appears to just be from `forwardRef`.

**Comment by ljharb** [comment]
This appears to be fixed, presumably by ddb06277771b82919b76e2d46582533522ab1bbd (which fixed #2146). When I update your repro repo to the latest enzyme and react adapter, everything passes.

**Comment by ljharb** [comment]
@Izkata if you're still having trouble on the latest version of everything, please file a new issue.
